playwright-e2e/pages/page-base.ts

```diff
@@ -0,0 +1,107 @@
+import { Locator, Page, Response } from '@playwright/test';
+import { PageInterface } from './page-interface';
+
+export default abstract class PageBase implements PageInterface {
+  protected readonly page: Page;
+  protected readonly baseUrl: string;
+
+  protected constructor(page: Page, baseURL: string) {
+    this.page = page;
+    this.baseUrl = baseURL;
+  }
```

Comment:
**_#minor_**

Can be shortened using [typescript constructor shorthand](https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd).